src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -329,9 +329,14 @@ private static boolean isValidFindSalaryRange(String salaryRange) {
         }
         String upperBound = salaryRange.split(" - ", 2)[1];
         String lowerBound = salaryRange.split(" - ", 2)[0];
-        long upperBoundLong = SalaryParserUtil.parseStringToLong(upperBound);
-        long lowerBoundLong = SalaryParserUtil.parseStringToLong(lowerBound);
-        if (upperBoundLong > Salary.MAXIMUM_SALARY_LONG || lowerBoundLong > upperBoundLong) {
+        try {
+            long upperBoundLong = SalaryParserUtil.parseStringToLong(upperBound);
+            long lowerBoundLong = SalaryParserUtil.parseStringToLong(lowerBound);
+            if (upperBoundLong > Salary.MAXIMUM_SALARY_LONG || lowerBoundLong > upperBoundLong) {
+                return false;
+            }
+        }
+        catch (NumberFormatException e) {
```

Comment:
Should we modify `SalaryParserUtil#parseStringToLong` to catch `NumberFormatException` and then throw `ParseException` instead? I just realised that the current way of handling `NumberFormatException` might have revealed the implementation of `SalaryParserUtil#parseStringToLong`

Reply:
Will fix in a later pr